### PR TITLE
fix(dynamic-currency-field): update styling and add ghost input

### LINKED ISSUE #143: UI bug: Fix visual cutoffs in pay/request view
screenshot from 2017-12-27 18 43 34

### PR DESCRIPTION
**Pull request by FrimJo**
Update both PayForm and RequestForm with a <span>-tag, which uses the same value as the input field and the input-field uses the width of the <span>-tag to get the correct size. Together with removing of the pinwheel (#172) it will be perfect. This fixes #143 

Also, update RequestForm to a React Component class, and changed some styling for the containers to present the content correctly.

before:
skarmavbild 2018-01-12 kl 09 39 49
skarmavbild 2018-01-12 kl 09 38 12

after:
skarmavbild 2018-01-12 kl 09 39 01
skarmavbild 2018-01-12 kl 09 38 55

### CODE REVIEW

File: app/components/Form/PayForm.js [hunk 1]
```diff
@@ -8,16 +8,31 @@ import CurrencyIcon from 'components/CurrencyIcon'
...
+  componentDidUpdate(prevProps, prevState) {
+    const { isOnchain, isLn, payform: { payInput }, fetchInvoice } = this.props
 
     // If on-chain, focus on amount to let user know it's editable
-    if (isOnchain) { this.amountInput.focus() }
+    if (isOnchain) {
+      this.amountInput.focus()
+    }
+
+    // If the with of the ghost input has changed, update the scrollWith
```
Comment (JimmyMow) [inline app/components/Form/PayForm.js]: width**

small typo 😉

**Comment by JimmyMow** [comment]
Looks good @FrimJo!

Looks like there are some conflicts and 1 tiny typo. Awesome PR 👍

**Comment by JimmyMow** [comment]
@FrimJo looks like there are a few lint errors. I'd love to get this merged once those are fixed 😄

**Comment by FrimJo** [comment]
Thank you @JimmyMow I’ll look into it first thing on Monday!

**Comment by JimmyMow** [comment]
@FrimJo any update on this PR?

**Comment by FrimJo** [comment]
Yea, it's on its way. I have just been busy at work, I'll try have it ready by Thursday evening (EST) :)

**Comment by thinkjanis** [comment]
Any update on this @FrimJo?

**Comment by FrimJo** [comment]
Sorry, I have fixed all lint errors except two. They're about not using setState in didUpdate and didMount which eslint doesn't approve. I read in the official docs that using setState in didMount is okay if getting a DOM value is necessary, which it is in this PR, but haven't found a work around for not using setState in didUpdate.

**Comment by thinkjanis** [comment]
So this is fixed in the new version?

**Comment by FrimJo** [comment]
Sorry, it is not. I removed the branch because I didn’t manage to fix the lint errors, I still have the code though. I haven’t had the time to look in to it any further at the moment.

**Pull request closed without merging**